src/main/java/br/unicamp/cst/core/entities/Memory.java

```diff
@@ -79,5 +79,11 @@
 	 */
 	public Long getTimestamp();
 	
+	/**
+	 * Add a memory observer to its list
+	 * @param memoryObserver
+	 */
+	public void addMemoryObservers(MemoryObserver memoryObserver);
```

Comment:
please remove the 's' as this method only adds a single memoryObserver

```suggestion
	public void addMemoryObserver(MemoryObserver memoryObserver);
```